AK/Optional.h

```diff
@@ -307,7 +320,27 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> : public Optiona
     }
 
 private:
-    alignas(T) u8 m_storage[sizeof(T)];
+    ALWAYS_INLINE constexpr void construct_null_if_necessairy(bool should_construct = is_constant_evaluated())
```

Comment:
I'm surprised this works. it was my impression that arguments were never constant expressions.